conn.go

```diff
@@ -56,7 +64,10 @@ func (c *clientConn) loop() {
 	err := c.recv()
 	if err != nil {
 		c.broadcastErr(err)
-		c.errCh <- err
+		c.errCond.L.Lock()
+		c.err = err
+		c.errCond.Broadcast()
```

Comment:
It would also probably make sense to have the `c.err` and boadcasting of a signal that it is now set in the `broadcastErr` method?